### ISSUE TITLE
Springboot 3- VCAP Services sealed secrets issue for Xsuaa

### ISSUE THREAD

**Issue by Pavan89851** [SAP/cloud-security-services-integration-library#1294]
**Issue Summary**
- A clear and concise description of the issue.

When we upgrading spring boot 2 to 3 with the dependency of cloud security xsuaa with the version `<version.xsuaa-java-sbom>3.1.0</version.xsuaa-java-sbom>` we are facing below error message when deploying the application in the container.
`{"@version":1,"source_host":"...........","message":"\n\nError starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.","thread_name":"main","@timestamp":"2023-09-22T10:59:16.624+0000","level":"INFO","logger_name":"org.springframework.boot.autoconfigure.logging.ConditionEvaluationReportLogger"}
{"exception":{"exception_class":"org.springframework.beans.factory.BeanDefinitionStoreException","exception_message":"Failed to process import candidates for configuration class [com.sap..................]: A JSONArray text must start with '[' at 1 [character 2 line 1]","stacktrace":"org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [com.sap...........]`

- Steps to reproduce the problem: 1. 2. 3.
when we provide the environment variable as VCAP_SERVICES and passing the secrets like below it works fine without any issues:
 **secrets: |
      {"xsuaa": [{"credentials": { "clientid": "dummy","xsappname": "dummy","clientsecret": "dummy","url": "dummy","uaadomain": "dummy"}}]} 

But when we tried with the sealed secrets by passing all those values with volume mounts and referring from the environment of that sealed secret name it was giving me the exception and end up with the Crash loop back off error.**

**Issue Report Checklist**

- [ ] Real, current bug
- [ ] Not a duplicate  
- [ ] Not covered in "Common Pitfalls" section of corresponding module's README.md e.g. [java-security#common-pitfalls](https://github.com/SAP/cloud-security-services-integration-library/tree/main/java-security#common-pitfalls)
- [ ] Reproducible
- [ ] Good summary
- [ ] Well-documented 
    - [ ] log level increased to `DEBUG` debug logs provided
    - [ ] POM provided
    - [ ] dependency tree provided
    - [ ] code snippet provided (when applicable)

**Comment by finkmanAtSap** [comment]
Hi,

We are using https://github.com/SAP/btp-environment-variable-access/tree/main to read the service credentials.
Your exception message contains no details at all which classes are involved, so I cannot say if it is a class from our repository that throws.
However, I think it might be more related to btp-environment-variable-access or how you are supplying the credentials.
Can you please check their guide for Kubernetes and make sure you are doing everything as described?

**Comment by Pavan89851** [comment]
In the above error it was throwing exception with the spring boot main class of the application.
since reference-instance plan is not supporting with spring-xsuaa version which I have mentioned above is not working with the env variable as "SERVICE_BINDING_ROOT". so, we are trying to pass all those values as sealed secrets like client id and client secret etc. as a workaround.
In that perspective what would be the Environment variables of client id and client secret it will expect to parse the xsuaa properties or read the values from VCAP for spring-xsuaa with <version.xsuaa-java-sbom>3.1.0</version.xsuaa-java-sbom>

**Comment by liga-oz** [comment]
let me rephrase your issue.
You are running on k8s and want to read the Xsuaa service configuration from VCAP_SERVICES environment variable?

**Comment by Pavan89851** [comment]
Yes I am running on k8s and want to read the Xsuaa service configuration through yaml format but not with Json format.

**Comment by liga-oz** [comment]
if you want to use yaml as the service configuration property source then it has to follow the following format https://github.com/SAP/cloud-security-services-integration-library/blob/main/samples/spring-security-hybrid-usage/src/test/resources/application.yml

**Comment by Pavan89851** [comment]
That is for spring-security, isn't it? 
What if I want to pass and get the environment variables for spring-xsuaa?

**Comment by liga-oz** [comment]
yes for spring-security, for spring-xsuaa it's just `xsuaa` prefix see here: https://github.com/SAP/cloud-security-services-integration-library/blob/main/spring-security/Migration_SpringXsuaaProjects.md#value

**Comment by liga-oz** [comment]
closing due to inactivity
